compiler/p/codegen/PPCAOTRelocation.cpp

```diff
@@ -39,6 +39,15 @@ void TR::PPCPairedRelocation::mapRelocation(TR::CodeGenerator *cg)
    {
    if (cg->comp()->getOption(TR_AOT))
       {
+      // The validation and inlined method ExternalRelocations are added after
+      // binary encoding is finished so that their actual relocation records
+      // will end up at the start. This way other relocations can depend on the
+      // appropriate validations already having completed.
+      //
+      // This relocation may also need to depend on previous validations, but
+      // those are already in the AOT relocation list by this point. Add it at
+      // the beginning to maintain the correct relative ordering.
+
```

Comment:
This is a little unclear to me. You've added a mechanism to allow callers of `addAOTRelocation()` to specify where to place the relocation, and this is the only place that uses `TR::AOTRelocationAtFront`, but the comment suggests that it doesn't really need it. Is this mechanism actually necessary? If not, is there any value in introducing it?

Reply:
The `ExternalRelocation`s get (roughly) reversed in the process of grouping them into `IteratedExternalRelocation`s, so adding this one at the end (like usual) will cause its relocation record to appear at the very beginning, before the validations. Adding it at the beginning here will put the relocation record at the end.

Reply:
OK, sounds reasonable, though it's a little odd that this is only needed on Power. How does this manage to work on other platforms? Wouldn't they have the same sort of ordering constraints on dependent relocations? Or are class address and validation relocations not dependent on other platforms?

Reply:
`ExternalRelocation`s are typically created earlier, during binary encoding. `TR::PPCPairedRelocation`s (which of course are specific to Power) are generated during evaluation instead, so their locations are `TR::Instruction*` rather than pointers into the binary code. They're translated into `ExternalRelocation` here, after binary encoding is complete, but also after the validations have been generated.

Reply:
Fair enough, thanks. Shame it has to be yet another platform-specific oddity.

LGTM.